spring-boot-project/spring-boot/src/main/java/org/springframework/boot/logging/LoggingSystemProperties.java

```diff
@@ -20,11 +20,12 @@
 import java.nio.charset.StandardCharsets;
 import java.util.function.BiConsumer;
 
+import org.springframework.boot.context.properties.source.ConfigurationPropertySources;
```

Comment:
The current `logging` package does not allow the content of the `context` package to be imported, I am not sure if I need to modify the rules or use other methods to create property resolver.

https://github.com/spring-projects/spring-boot/blob/2b5bd71eb46b9f7295e195b3db5b7fa27ce9b205/src/checkstyle/import-control.xml#L80

Reply:
That restriction was introduced in https://github.com/spring-projects/spring-boot/issues/8611. I'm not sure if we should relax it or not. I wonder if we can get away with using the `Environment` directly since it's already a `PropertyResolver`.

Reply:
Thank you for your reply! If you use `Environment` directly, we may need to set its `ignoreUnresolvableNestedPlaceholders` property to `true` to ensure that it is consistent with the original behavior. Since the `Environment` is modified, this will also affect the external use of it. Maybe we should make a copy of `Environment`?

Reply:
Good point. I need to think about this one for a bit.